dataset/src/main/scala/frameless/FramelessSyntax.scala

```diff
@@ -7,7 +7,8 @@ trait FramelessSyntax {
     def typed[T, U: TypedEncoder]: TypedColumn[T, U] = new TypedColumn[T, U](self)
   }
 
-  implicit class DatasetSyntax[T: TypedEncoder](self: Dataset[T]) {
-    def typed: TypedDataset[T] = TypedDataset.create[T](self)
+  implicit class DatasetSyntax[T](self: Dataset[T]) {
+    def typed(implicit ev: TypedEncoder[T]): TypedDataset[T] = TypedDataset.create[T](self)
+    def toTyped[U: TypedEncoder]: TypedDataset[U] = TypedDataset.create(self.as(TypedExpressionEncoder[U]))
```

Comment:
Given that `T` is ignored here, I wonder if it should be given a scarier name or be restricted to `T = Row`